File: Accelbuffer/Accelbuffer/Enums/ObjectType.cs
namespace Accelbuffer
{
    /// <summary>
    /// 表示一个对象的数据类型（4位）
    /// </summary>
    internal enum ObjectType
    {
        /// <summary>
        /// 丢失的类型
        /// </summary>
        Missing = 0,

        /// <summary>
        /// 8位固定长度类型
        /// </summary>
        Fixed8 = 1,

        /// <summary>
        /// 16位固定长度类型
        /// </summary>
        Fixed16 = 2,

        /// <summary>
        /// 24位固定长度类型
        /// </summary>
        Fixed24 = 3,

        /// <summary>
        /// 32位固定长度类型
        /// </summary>
        Fixed32 = 4,

        /// <summary>
        /// 40位固定长度类型
        /// </summary>
        Fixed40 = 5,

        /// <summary>
        /// 48位固定长度类型
        /// </summary>
        Fixed48 = 6,

        /// <summary>
        /// 56位固定长度类型
        /// </summary>
        Fixed56 = 7,

        /// <summary>
        /// 64位固定长度类型
        /// </summary>
        Fixed64 = 8,

        /// <summary>
        /// 72位固定长度类型
        /// </summary>
        Fixed72 = 9,

        /// <summary>
        /// 80位固定长度类型
        /// </summary>
        Fixed80 = 10,

        /// <summary>
        /// 88位固定长度类型
        /// </summary>
        Fixed88 = 11,

        /// <summary>
        /// 96位固定长度类型
        /// </summary>
        Fixed96 = 12,

        /// <summary>
        /// 104位固定长度类型
        /// </summary>
        Fixed104 = 13,

        /// <summary>
        /// 128位固定长度类型
        /// </summary>
        Fixed128 = 14,

        /// <summary>
        /// 前缀长度类型（Variant字节长度 + N字节数据）
        /// </summary>
        LengthPrefixed = 15
    }
}

File: Accelbuffer/Accelbuffer.Java/src/accelbuffer/core/ObjectType.java
package accelbuffer.core;

/**
 * 表示一个对象的数据类型（4位）
 */
enum ObjectType
{
    /**
     * 丢失的类型
     */
    MISSING(0),
    /**
     * 8位固定长度类型
     */
    FIXED8(1),
    /**
     * 16位固定长度类型
     */
    FIXED16 (2),
    /**
     * 24位固定长度类型
     */
    FIXED24(3),
    /**
     * 32位固定长度类型
     */
    FIXED32(4),
    /**
     * 40位固定长度类型
     */
    FIXED40(5),
    /**
     * 48位固定长度类型
     */
    FIXED48(6),
    /**
     * 56位固定长度类型
     */
    FIXED56(7),
    /**
     * 64位固定长度类型
     */
    FIXED64(8),
    /**
     * 72位固定长度类型
     */
    FIXED72(9),
    /**
     * 80位固定长度类型
     */
    FIXED80(10),
    /**
     * 88位固定长度类型
     */
    FIXED88(11),
    /**
     * 96位固定长度类型
     */
    FIXED96(12),
    /**
     * 104位固定长度类型
     */
    FIXED104(13),
    /**
     * 128位固定长度类型
     */
    FIXED128(14),
    /**
     * 前缀长度类型（Variant字节长度 + N字节数据）
     */
    LENGTH_PREFIXED(15);

    private int value;

    private ObjectType(int value)
    {
        this.value = value;
    }

    public int getValue()
    {
        return value;
    }
}
